### PR TITLE
implements structural changes and adds config topic

### PR DESCRIPTION
**Pull request by dedemorton**
I've added a config topic to address the concern that @tbragin expressed here https://github.com/elastic/libbeat/issues/294. Most of the other changes are minor (sorry about the whitespace changes).

### CODE REVIEW

File: docs/configuration.asciidoc [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+Before modifying configuration settings, make sure you've read
+<<topbeat-configuration>>.
+
+The Topbeat configuration file uses http://yaml.org/[YAML] for its syntax.
+The file consists of the following sections:
+
+* <<configuration-input>>
+* {libbeat}/configuration.html#configuration-output[Output]
+* {libbeat}/configuration.html#configuration-shipper[Shipper]
+* {libbeat}/configuration.html#configuration-logging[Logging]
```
Comment (monicasarbu) [inline docs/configuration.asciidoc]: * Run Options (Optional)
* Input (sounds a bit strange, but this is the same of the section)

Comment (monicasarbu) [inline docs/configuration.asciidoc]: * Logging (Optional)

**Comment by dedemorton** [comment]
@monicasarbu OK, I've changed the intro to be very specific about how things are currently organized and where they are documented. I will apply this change across the other beats to be very explicit. Until we have some kind of strategy for reuse, I think this approach will work, and it will help educate users about the Platform Reference as a source of common information about Beats.

**Comment by monicasarbu** [comment]
@dedemorton +1

**Pull request merged**